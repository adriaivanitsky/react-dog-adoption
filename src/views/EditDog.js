import React from 'react';
import Header from '../components/Header';
import Form from '../components/Form';
import { useState, useEffect } from 'react';
import { getDogById } from '../services/dogData';
import { updateDog } from '../services/dogData';

export default function EditDog(props) {
  const [loading, setLoading] = useState(true);
  const [name, setName] = useState('');
  const [age, setAge] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const [breed, setBreed] = useState('');
  const [success, setSuccess] = useState('');
  const id = props.match.params.id;
  //useEffect for preloading dogdata so the form can already be filled out
  //almost identical to admin as far as save function but inside of save function call updateDog
  //instead of createDog().

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogById(id);
      setName(data.name);
      setAge(data.age);
      setBio(data.bio);
      setImage(data.image);
      setBreed(data.breed);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  const save = async (e) => {
    try {
      e.preventDefault();
      await updateDog({ id, name, age, bio, image, breed });
      setSuccess('dog successfully updated');
    } catch {
      setSuccess('sorry! something went wrong. please try again');
    }
  };

  return (
    <>
      <Header />
      <Form
        {...{
          loading,
          setLoading,
          name,
          setName,
          age,
          setAge,
          bio,
          setBio,
          image,
          setImage,
          breed,
          setBreed,
          success,
          save,
        }}
      />
    </>
  );
}
