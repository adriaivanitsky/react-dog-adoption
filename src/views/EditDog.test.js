import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import EditDog from './EditDog.js';

test('editdog should render dog information to a form', async () => {
  const { container } = render(
    <MemoryRouter>
      <EditDog match={{ params: { id: 5 } }} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('pumpkin');
  expect(container).toMatchSnapshot();
});
