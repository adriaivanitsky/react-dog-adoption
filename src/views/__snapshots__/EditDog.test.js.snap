// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editdog should render dog information to a form 1`] = `
<div>
  <header>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      home 
    </a>
    <a
      href="/admin"
    >
       admin
    </a>
  </header>
  <form>
    <p />
    <label>
      name
      <input
        placeholder="name here"
        type="text"
        value="pumpkin"
      />
    </label>
    <label>
      age
      <input
        placeholder="age"
        type="number"
        value="6"
      />
    </label>
    <label>
      bio
      <input
        placeholder="bio"
        type="text"
        value="fun loving pup who loves long walks and needs lots of attention "
      />
    </label>
    <label>
      image
      <input
        type="text"
        value="https://thedogman.net/wp-content/uploads/2020/09/Australian-Shepherd-3.jpg?x47951"
      />
    </label>
    <label>
      breed
      <input
        placeholder="breed"
        type="text"
        value="australian shepherd "
      />
    </label>
    <button>
      save
    </button>
  </form>
</div>
`;
